esmvalcore/_data_finder.py

```diff
@@ -67,16 +69,35 @@ def get_start_end_year(filename):
         filename_list[ind] for ind, _ in enumerate(pos_ydates)
         if pos_ydates_r[ind] or pos_ydates_l[ind]
     ]
-
+    start_year = None
+    end_year = None
     if len(dates) == 1:
         start_year = int(dates[0][:4])
         end_year = start_year
     elif len(dates) == 2:
         start_year, end_year = int(dates[0][:4]), int(dates[1][:4])
     else:
-        raise ValueError('Name {0} dates do not match a recognized '
-                         'pattern'.format(name))
-
+        # Slower than just parsing the name
+        try:
+            cubes = iris.load(filename)
+        except OSError:
+            raise ValueError('File {0} can not be read'.format(filename))
+
+        for cube in cubes:
+            logger.debug(cube)
+            try:
+                time = cube.coord('time')
+            except iris.exceptions.CoordinateNotFoundError:
+                continue
+            start_year = time.cell(0).point.year
+            end_year = time.cell(-1).point.year
+            break
+
+    if not start_year or not end_year:
```

Comment:
```suggestion
    if start_year is None or end_year is None:
```
or this will also fail if the year is 0